pkg/kwokctl/runtime/kind/cluster.go

```diff
@@ -668,6 +669,91 @@ func (c *Cluster) LogsFollow(ctx context.Context, name string, out io.Writer) er
 	return c.logs(ctx, name, out, true)
 }
 
+// CollectLogs returns the logs of the specified component.
+func (c *Cluster) CollectLogs(ctx context.Context, name string, dir string) error {
+	logger := log.FromContext(ctx)
+
+	conf, err := c.Config(ctx)
+	if err != nil {
+		return err
+	}
+
+	var errs []error
+
+	componentsDir := filepath.Join(dir, "components")
+	for _, component := range conf.Components {
+		path := filepath.Join(componentsDir, component.Name+".log")
+		f, err := file.Open(path, 0640)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		err = c.Logs(ctx, component.Name, f)
+		if err != nil {
+			logger.Error("Failed to get log", err)
+			err = f.Close()
+			if err != nil {
+				err = os.Remove(path)
+				if err != nil {
+					logger.Error("Failed to remove file", err)
+				}
+			}
+			continue
+		}
+		err = f.Close()
+		if err != nil {
+			logger.Error("Failed to close file", err)
+			err = os.Remove(path)
+			if err != nil {
+				logger.Error("Failed to remove file", err)
+			}
+		}
+	}
+
+	if conf.Options.KubeAuditPolicy != "" {
+		filePath := filepath.Join(componentsDir, "audit.log")
+		f, err := file.Open(filePath, 0640)
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			err = c.AuditLogs(ctx, f)
+			if err != nil {
+				logger.Error("Failed to get audit log", err)
+				errs = append(errs, err)
+			}
+			err = f.Close()
+			if err != nil {
+				logger.Error("Failed to close file", err)
+				err = os.Remove(filePath)
+				if err != nil {
+					logger.Error("Failed to remove file", err)
+				}
+			}
+		}
+	}
+
+	kindPath, err := c.preDownloadKind(ctx)
+	if err != nil {
+		errs = append(errs, err)
+		return errors.Join(errs...)
+	}
```

Comment:
Move to before of dump logs
```suggestion
	kindPath, err := c.preDownloadKind(ctx)
	if err != nil {
		return err
	}
```